cmake/kokkos_cxx.cmake

```diff
@@ -211,6 +188,37 @@ ELSE()
   ENDIF()
 ENDIF()
 
+# Enforce that extensions are turned off for nvcc_wrapper.
+# For compiling CUDA code using nvcc_wrapper, we will use the host compiler's
+# flags for turning on C++11.  Since for compiler ID and versioning purposes
+# CMake recognizes the host compiler when calling nvcc_wrapper, this just
+# works.  Both NVCC and nvcc_wrapper only recognize '-std=c++11' which means
+# that we can only use host compilers for CUDA builds that use those flags.
+# It also means that extensions (gnu++11) can't be turned on for CUDA builds.
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
+  IF(NOT DEFINED CMAKE_CXX_EXTENSIONS)
+    MESSAGE(FATAL_ERROR "NVCC doesn't support C++ extensions.  Set -DCMAKE_CXX_EXTENSIONS=OFF")
+  ELSEIF(CMAKE_CXX_EXTENSIONS)
+    MESSAGE(FATAL_ERROR "NVCC doesn't support C++ extensions.  Set -DCMAKE_CXX_EXTENSIONS=OFF")
+  ENDIF()
```

Comment:
```CMake
if(NOT DEFINED CMAKE_CXX_EXTENSIONS OR CMAKE_CXX_EXTENSIONS)
```

Reply:
Done

Reply:
I would definitely change this to a warning for cmake extensions not being defined.